Add shell script to scan for IOCs via SSM

dav3r opened the pull request:
# <!-- Use the title to describe PR changes in the imperative mood --> #

## 🗣 Description ##

<!-- Describe the "what" of your changes in detail. -->
<!-- To avoid scope creep, limit changes to a single goal. -->

This PR adds an "extras" directory containing a shell script that can be used to scan a list of AWS EC2 instances via [SSM](https://docs.aws.amazon.com/systems-manager/latest/userguide/what-is-systems-manager.html) for indicator of compromise strings.

## 💭 Motivation and context ##

This script is useful for cases where the instances you want to scan are only accessible via SSM.

<!-- Why is this change required? -->
<!-- What problem does this change solve? How did you solve it? -->
<!-- Mention any related issue(s) here using appropriate keywords such -->
<!-- as "closes" or "resolves" to auto-close them on merge. -->

## 🧪 Testing ##

I ran the script with a variety of inputs and verified that it worked as expected.  I also tested the script with no arguments, with invalid arguments (e.g. a non-existent IOC file), and with `-h` and with `--help` and confirmed that the usage text was displayed as intended in those cases.

<!-- How did you test your changes? How could someone else test this PR? -->
<!-- Include details of your testing environment, and the tests you ran to -->
<!-- see how your change affects other areas of the code, etc. -->

<!--
## 📷 Screenshots (if appropriate) ##

Uncomment this section if a screenshot is needed.

-->

## ✅ Pre-approval checklist ##

<!-- Remove any of the following that do not apply. -->
<!-- Draft PRs should have one or more unchecked boxes. -->
<!-- If you're unsure about any of these, don't hesitate to ask. -->
<!-- We're here to help! -->

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
- [x] All new and existing tests pass.

## ✅ Pre-merge checklist ##

<!-- Remove any of the following that do not apply. -->
<!-- These boxes should remain unchecked until the pull request has been -->
<!-- approved. -->

- [x] Finalize version.

## ✅ Post-merge checklist ##

<!-- Remove any of the following that do not apply. -->

- [x] Create a release.

dav3r committed:
Add script to scan AWS instances for IOCs via SSM

dav3r committed:
Update README to mention extras/ioc_scan_by_host.sh

dav3r committed:
Bump version from 1.5.0 to 1.5.1

jsf9k commented on extras/ioc_scan_by_host.sh:
```diff
@@ -0,0 +1,63 @@
+# and IOC file); the rest are the instance IDs.
+instances=("${@:2}")
+
+echo "IOC List is: \"${iocList[*]}\""
+echo "Instances are: \"${instances[*]}\""
```
`echo` doesn't require that the text to be printed be in quotes, so I think there are several places in this script where you can simplify your quoting by removing the outermost double quotes, e.g.:
```suggestion
echo IOC List is: "${iocList[*]}"
echo Instances are: "${instances[*]}"
```

dav3r replied:
Yup, thanks for that reminder!  See https://github.com/cisagov/ioc-scanner/pull/45/commits/b1c498908f054a69840e5ac94b40a5aeec4967fb.

jsf9k reviewed (commented):
I have one comment that applies all over the show in `ioc_scan_by_host.sh`.  Let me know if you agree.

dav3r committed:
Remove unnecessary outer double quotes from echo statements

Also, clean up an echo that includes a newline.

Co-authored-by: Shane Frasier <jeremy.frasier@trio.dhs.gov>

jsf9k commented on extras/ioc_scan_by_host.sh:
```diff
@@ -0,0 +1,63 @@
+
+# Loop through all instance IDs
+for instance_id in "${instances[@]}"; do
+  # "echo -e" allows us to use \n for newlines.
+  echo -e "\nSearching $instance_id:"
```
This is the usual way of doing it.
```suggestion
  echo
  echo Searching "$instance_id":
```

dav3r replied:
Works for me!

jsf9k reviewed (approved):
I have one minor suggestion, but my approval stands even if you choose to disregard it.

dav3r committed:
Use a more common way to echo a newline

Co-authored-by: Shane Frasier <jeremy.frasier@trio.dhs.gov>

mcdonnnj commented on README.md:
````diff
@@ -82,6 +82,30 @@ ansible --inventory=hosts-file cool-servers \
+```console
+$ ./extras/ioc_scan_by_host.sh my-ioc-strings.txt i-0123456789abcdef0
+
+IOC List is: "192.168.1[.]1 sketchy-site[.]org IOC_HASH_1 IOC_HASH_2"
````
Just as a discussion point does mentioning hashes make sense here? The Python logic in this project generates hashes of files and checks those while this script searches for strings in log files. I don't want to conflate the functionality of the two components as the script added in this PR does no hash generation. Maybe something more like:

```suggestion
IOC List is: "192.168.1[.]1 sketchy-site[.]org IOC_STRING_1 IOC_STRING_2"
```

dav3r replied:
That is an excellent point.  I will make that change shortly.

mcdonnnj commented on README.md:
```diff
@@ -82,6 +82,30 @@ ansible --inventory=hosts-file cool-servers \
+Starting session with SessionId: iam.username-0123456789abcdef0
+0 found for 192.168.1[.]1
+0 found for sketchy-site[.]org
+0 found for IOC_HASH_1
+0 found for IOC_HASH_2
```
If the above suggestion is accepted this should be as well.

```suggestion
0 found for IOC_STRING_1
0 found for IOC_STRING_2
```

dav3r committed:
Modify wording to avoid confusion

This script scans for files containing IOC strings and NOT file hashes, so it's best to avoid mentioning anything about hashes in the documentation for this script.

Co-authored-by: Nick <50747025+mcdonnnj@users.noreply.github.com>

mcdonnnj reviewed (approved):
:shipit:
